frameworks/AutoGluon/exec.py

```diff
@@ -46,22 +44,14 @@ def run(dataset, config):
     is_classification = config.type == 'classification'
     training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
 
-    load_raw = config.framework_params.get('_load_raw', False)
-    if load_raw:
-        train, test = load_data_raw(dataset=dataset)
-    else:
-        column_names, _ = zip(*dataset.columns)
-        column_types = dict(dataset.columns)
-        train = pd.DataFrame(dataset.train.data, columns=column_names).astype(column_types, copy=False)
-        log.info(f"Columns dtypes:\n{train.dtypes}")
-        test = pd.DataFrame(dataset.test.data, columns=column_names).astype(column_types, copy=False)
-
-    del dataset
```

Comment:
Are there ways to ensure the data is only present once in memory? This was my attempt previously to do this, but I suspect it actually exists multiple times in memory. A good test would be to see if AutoGluon still crashes on `KDDCup09-Upselling` from OOM before getting to AG code on an m5.2xlarge. If the duplicate instances can be eliminated, this would free up a lot of memory for AutoML systems to work with when the datasets are large.

Reply:
running the benchmark with `--profiling` option will provide with some additional logs, especially for main memory changes when loading data.
right before calling the subprocess running the client code, and just after  having rerialized the dataframes, I call 
```
dataset.release()
```
to ensure than the data doen't remain in memory.

Then, on the "callee" side, the data are just pickled and if your code doesn't manipulate the frames (it should not be needed anymore, now that we automatically provide pd.DataFrame with the correct dtypes), then I don't see any reason for duplications.

Reply:
ok, sounds great!